test/e2e/test_real_devices.py

```diff
@@ -238,10 +193,8 @@ def test_headers_in_result_devices(self):
         job = self.real_device_backend.run(
             transpile(qc1, backend=self.real_device_backend), header=custom_header
         )
-        job.wait_for_final_state(wait=300, callback=self.simple_job_callback)
-        result = job.result()
+        job.wait_for_final_state()
         self.assertTrue(custom_header.items() <= job.header().items())
-        self.assertTrue(custom_header.items() <= result.header.to_dict().items())
```

Comment:
Why removing this given the test is to ensure qobj headers are passed in the result?

Reply:
you're right, I was testing on simulator causing this to fail - added back in [cd1ac42](https://github.com/Qiskit/qiskit-ibm-provider/pull/438/commits/cd1ac42ffeb76472a24ae3189117c05a6f27b4e8)